gcsio/src/test/java/com/google/cloud/hadoop/gcsio/BatchHelperTest.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.hadoop.gcsio;
+
+import static com.google.cloud.hadoop.gcsio.GoogleCloudStorageTestUtils.BUCKET_NAME;
+import static com.google.cloud.hadoop.gcsio.GoogleCloudStorageTestUtils.JSON_FACTORY;
+import static com.google.cloud.hadoop.gcsio.GoogleCloudStorageTestUtils.OBJECT_NAME;
+import static com.google.cloud.hadoop.gcsio.GoogleCloudStorageTestUtils.metadataResponse;
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.api.client.googleapis.batch.json.JsonBatchCallback;
+import com.google.api.client.googleapis.json.GoogleJsonError;
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpHeaders;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.testing.http.MockHttpTransport;
+import com.google.api.client.util.DateTime;
+import com.google.api.services.storage.Storage;
+import com.google.api.services.storage.model.StorageObject;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.Date;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class BatchHelperTest {
+
+  private BatchHelper.Factory batchFactory;
+
+  @Before
+  public void setUp() {
+    batchFactory = new BatchHelper.Factory();
+  }
+
+  @Test
+  public void allRequestsAreSentInSingleBatch() throws IOException {
```

Comment:
Can we make the test more readable, either add comments or refactor. I prefer the following pattern when writing test cases:

setup(x, y, z...)
run(x, y, z...)
verify(x, y, z...)

it should be easy to see the cause and effect between setup/run and verify, e.g., I setup batch size as 2, and I added 2 request, so I expect 1 batch request.

Reply:
Added comments.

There are 3 requests because requests interceptor intercepts not only single batch request, but also individual requests inside the batch request.